### PR TITLE
More integration tests for Model + some fixes in the framework

### PR DESCRIPTION
**Pull request by mwielgus**
Fix for #487

### CODE REVIEW

File: integration/heapster_api_test.go [hunk 1]
```diff
@@ -53,16 +54,32 @@ var (
 
 func deleteAll(fm kubeFramework, ns string, service *kube_api.Service, rc *kube_api.ReplicationController) error {
 	glog.V(2).Infof("Deleting rc %s/%s...", ns, rc.Name)
-	if err := fm.DeleteRC(ns, rc); err != nil {
-		glog.V(2).Infof("Failed to delete rc: %v", err)
-		return err
+	retries := 5
+	for {
```
Comment (piosz) [inline integration/heapster_api_test.go]: How about using here time wait pattern here (see link below)?

https://github.com/kubernetes/kubernetes/blob/master/test/e2e/util.go#L394

File: integration/heapster_api_test.go [hunk 2]
```diff
@@ -53,16 +54,32 @@ var (
...
+		retries--
 	}
 	glog.V(2).Infof("Deleted rc %s/%s.", ns, rc.Name)
 
 	glog.V(2).Infof("Deleting service %s/%s.", ns, service.Name)
-	if err := fm.DeleteService(ns, service); err != nil {
-		glog.V(2).Infof("Failed to delete service: %v", err)
-		return err
+	retries = 5
+	for {
```
Comment (piosz) [inline integration/heapster_api_test.go]: Same here.

File: integration/heapster_api_test.go [hunk 3]
```diff
@@ -419,34 +417,195 @@ func runSinksTest(fm kubeFramework, svc *kube_api.Service) error {
...
+}
+
 func runModelTest(fm kubeFramework, svc *kube_api.Service) error {
-	podList, err := fm.GetPodList()
+	podList, err := fm.GetRunningPods()
 	if err != nil {
 		return err
 	}
-	if len(podList.Items) == 0 {
+	if len(podList) == 0 {
```
Comment (piosz) [inline integration/heapster_api_test.go]: I'd create a dummy pod here rather than assume that at least one is running in the cluster.

Comment (mwielgus) [inline integration/heapster_api_test.go]: There is Heapster running ...

File: integration/heapster_api_test.go (hunk 1 above)
Comment (mwielgus) [inline integration/heapster_api_test.go]: This is a temporary fix for unreliable delete - see #559. Delete should mean delete and there should be no reason for retries. I will refactor this completely in the next PR.

File: integration/heapster_api_test.go (hunk 2 above)
Comment (mwielgus) [inline integration/heapster_api_test.go]: As above.

File: integration/heapster_api_test.go (hunk 3 above)
Comment (piosz) [inline integration/heapster_api_test.go]: Ok, let's assume it's enough.

File: integration/heapster_api_test.go (hunk 2 above)
Comment (piosz) [inline integration/heapster_api_test.go]: Ack

**Comment by piosz** [comment]
LGTM

**Comment by vishh** [comment]
LGTM. Thanks @mwielgus !

**Pull request merged**